src/Datadog.Trace/Util/EnvironmentHelpers.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Serilog;
+
+namespace Datadog.Trace.Util
+{
+    /// <summary>
+    /// Helpers to access environment variables
+    /// </summary>
+    internal static class EnvironmentHelpers
+    {
+        private static readonly ILogger Logger = DatadogLogging.GetLogger(typeof(EnvironmentHelpers));
+
+        /// <summary>
+        /// Safe wrapper around Environment.GetEnvironmentVariable
+        /// </summary>
+        /// <param name="key">Name of the environment variable to fetch</param>
+        /// <param name="defaultValue">Value to return in case of error</param>
+        /// <returns>The value of the environment variable, or the default value if an error occured</returns>
+        public static string GetEnvironmentVariable(string key, string defaultValue = null)
+        {
+            try
+            {
+                return Environment.GetEnvironmentVariable(key);
+            }
+            catch (Exception ex)
+            {
+                Logger.Warning(ex, "Error while reading environment variable {0}", key);
```

Comment:
This brings up an important question.

Is this a condition we should check for early in loading and bail out for?

Seems to me that we just won't be able to operate if we can't read environment variables.

Maybe I am wrong.

Reply:
Good point. @zacharycmontoya: if we don't load the managed profiler dll in the startup hook because `DD_DOTNET_TRACER_HOME` is not available, should we disable tracing entirely? Or is there still a change it can get loaded from the GAC if available? And if we ultimately don't load it, doesn't instrumentation already get disabled anyway?

For context, the goal here was to fix an issue where users ran child process in very restricted sandboxes, so managed code could not read env vars. Since env vars are optional for configuration (users can set config through other ways), we could just skip them instead of throwing an exception.

Reply:
Another note: reading env vars from native code is not a concern (at least not in this PR). In the case were we had this issue, the sandbox restrictions only applied to managed code. Native code could read env vars, which is how the profiler is loaded in the first place.

Reply:
Well the calling code does:

```var tracerHomeDirectory = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME") ?? string.Empty;```

So clearly, the thing was written with the assumption that DD_DOTNET_TRACER_HOME might not be defined. If not defined, it's going to fallback on the `netxxx` subfolder of the application working directory... Which is very unlikely to work.

Reply:
In that case, I think this change is good for this PR. Any further changes (bailing out early if we can't read env vars) is out of scope.